enhancements/kubelet/user-namespaces-support.md

```diff
@@ -0,0 +1,339 @@
+
+### Topology Considerations
+
+#### Hypershift / Hosted Control Planes
+
+N/A
+
+#### Standalone Clusters
+
+N/A
+
+#### Single-node Deployments or MicroShift
+
+N/A
+
+### Implementation Details/Notes/Constraints
+
+#### SecurityContextConstraint Updates
+
+##### ``UserNamespacesPolicy`` field
+
+Add the following to the [Openshift API](https://github.com/openshift/api/blob/76a71dac36a08eab1b240c6c8d4e39c813b1b12b/security/v1/types.go):
+
+```
+       // userNamespacesPolicy determines if the policy allows host users in containers.
+       // Valid values are "AllowHostNamespaces", "RequirePodNamespaces", and omitted.
+       // When omitted, this means no opinion and the platform is left to choose a reasonable default,
+       // which is subject to change over time.
+       // The current default is "AllowHostNamespaces".
+       // +openshift:enable:FeatureGate=UserNamespacesPodSecurityStandards
+       // +kubebuilder:validation:Enum="AllowHostNamespaces";"RequirePodNamespaces";""
+       // +optional
+       `UserNamespacesPolicy` NamespacesStrategyType `json:"userNamespacesPolicy" protobuf:"bytes,26,opt,name=userNamespacesPolicy"`
+...
+// NamespacesStrategyType shows the allowable values for the UserNamespacesPolicy field.
+type NamespacesStrategyType string
+...
+const (
+       // NamespacesStrategyTypeAllowHost allows a pod to set `hostUsers` field to either `true` or `false`
+       NamespacesStrategyTypeAllowHost NamespacesStrategyType = "AllowHostNamespaces"
+       // NamespaceStrategyRequirePod requires the `hostUsers` field be `false` in a pod.
+       NamespaceStrategyRequirePod NamespacesStrategyType = "RequirePodNamespaces"
+       // NamespaceStrategyEmpty will set to the default, which is currently "AllowHostNamespaces".
+       NamespaceStrategyEmpty NamespacesStrategyType = ""
+
+...
+)
+
+```
+
+This field branches from its fellow `AllowHost*` namespace fields to follow current API conventions, which maintain boolean fields are not extendable enough.
+
+##### restricted-v3
+
+This SCC profile will be identical to the existing restricted-v2, except it will set the `UserNamespacesPolicy` to `RequirePodNamespaces`,
+thus forcing pods to be in a user namespace. This will make it a more restrictive profile, as the user on the host will not be
+the same as the one inside the container.
+
+Note: upstream Kubernetes deemed `runAsUser` to be a safe field to use for pods with `hostUsers: false`. However, the philosophy
+of SCC utilizes defense-in-depth. `restricted-v3` will be the most restrictive policy.
+
+##### nonroot-v3
+
+This SCC profile will be similar to `nonroot-v2`, except it will set `UserNamespacesPolicy` to `RequirePodNamespaces`, thus forcing pods to be in a
+user namespace, and set the fields `runAsUser`/`runAsGroup`/`fsGroup` to `runAsAny` (in this way, it's similar to `anyuid` SCC, but without `allowPrivilegeEscalation` set to `true`).
```

Comment:
I think it would be valuable to emphasize that it offers more or less the same security as restricted-v2, while being like anyuid (except the mentioned `allowPrivilegeEscalation`).
Which means that it [allows `runAsAny`](https://kubernetes.io/docs/concepts/workloads/pods/user-namespaces/#integration-with-pod-security-admission-checks) and there is [no need to drop Capabilities](https://kubernetes.io/docs/concepts/workloads/pods/user-namespaces/#pods-and-userns).

Reply:
I looked it up in the enhancement, the capabilities are legit for baseline, but not restricted:

https://github.com/kubernetes/enhancements/blob/master/keps/sig-node/127-user-namespaces/README.md?plain=1#L476-L477

So in order to keep the level of `restricted-v2`, we would not allow any special capabilities?

So could this raise the need to another SCC that offers PSS baseline, with all capabilities? WDYT?